app/models/collection_object.rb

```diff
@@ -342,7 +342,7 @@ def self.in_geographic_item(geographic_item, limit, steps = false)
       retval = CollectionObject.where(id: step_4.sort)
     else
       retval = CollectionObject.joins(:geographic_items)
-        .where(GeographicItem.contained_by_where_sql(geographic_item.id))
+        .where(GeographicItem.within_union_of_sql(geographic_item.id))
```

Comment:
Perhaps unrelated but of interest.  There are union, intersect etc. generic methods in `Queries` and `Queries::Query` that might be of use.